bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -553,58 +567,15 @@ public void onMessage(byte[] buffer, int offset, int size)
     @Override
     public void unregistered(final SelectionKey key)
     {
+      handleSubscriberTeardown(key);
```

Comment:
Why the refactoring now? Do you see this method being used elsewhere in near future.